README.rst

```diff
@@ -385,51 +417,209 @@ Add ``openwisp_utils.loaders.DependencyLoader`` to ``TEMPLATES`` in your ``setti
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 For the purpose of showing an example, we added a simple "details" field to the
 `models of the sample app in the test project <https://github.com/openwisp/openwisp-firmware-upgrader/tree/master/tests/openwisp2/sample_firmware_upgrader/models.py>`_.
 
 You can add fields in a similar way in your ``models.py`` file.
 
-Extending the admin
+**Note**: for doubts regarding how to use, extend or develop models please refer to
+the `"Models" section in the django documentation <https://docs.djangoproject.com/en/dev/topics/db/models/>`_.
+
+8. Add swapper configurations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Once you have created the models, add the following to your ``settings.py``:
+
+.. code-block:: python
+
+    # Setting models for swapper module
+    FIRMWARE_UPGRADER_CATEGORY_MODEL = 'myupgrader.Category'
+    FIRMWARE_UPGRADER_BUILD_MODEL = 'myupgrader.Build'
+    FIRMWARE_UPGRADER_FIRMWAREIMAGE_MODEL = 'myupgrader.FirmwareImage'
+    FIRMWARE_UPGRADER_DEVICEFIRMWARE_MODEL = 'myupgrader.DeviceFirmware'
+    FIRMWARE_UPGRADER_BATCHUPGRADEOPERATION_MODEL = 'myupgrader.BatchUpgradeOperation'
+    FIRMWARE_UPGRADER_UPGRADEOPERATION_MODEL = 'myupgrader.UpgradeOperation'
+
+Substitute ``myupgrader`` with the name you chose in step 1.
+
+9. Create database migrations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Create and apply database migrations::
+
+    ./manage.py makemigrations
+    ./manage.py migrate
+
+For more information, refer to the
+`"Migrations" section in the django documentation <https://docs.djangoproject.com/en/dev/topics/migrations/>`_.
+
+10. Create the admin
 ~~~~~~~~~~~~~~~~~~~
 
-Please checkout the `admin.py file of the sample app <https://github.com/openwisp/openwisp-firmware-upgrader/tree/master/tests/openwisp2/sample_firmware_upgrader/admin.py>`_.
+Refer to the `admin.py file of the sample app <https://github.com/openwisp/openwisp-firmware-upgrader/tree/master/tests/openwisp2/sample_firmware_upgrader/admin.py>`_.
 
-You can change ``CategoryAdmin``, ``BuildAdmin`` and
-``BatchUpgradeOperationAdmin`` and these changes will be reflected in
-your admin interface.
+To introduce changes to the admin, you can do it in two main ways which are described below.
 
-Reusing the base tests
-~~~~~~~~~~~~~~~~~~~~~~
+**Note**: for more information regarding how the django admin works, or how it can be customized,
+please refer to `"The django admin site" section in the django documentation <https://docs.djangoproject.com/en/dev/ref/contrib/admin/>`_.
+
+1. Monkey patching
+##################
+
+If the changes you need to add are relatively small, you can resort to monkey patching.
+
+For example:
+
+.. code-block:: python
+
+    from openwisp_firmware_upgrader.admin import (  # noqa
+        BatchUpgradeOperationAdmin,
+        BuildAdmin,
+        CategoryAdmin,
+    )
+
+    BuildAdmin.list_display.insert(1, 'my_custom_field')
+    BuildAdmin.ordering = ['-my_custom_field']
+
+2. Inheriting admin classes
```

Comment:
While, doing this section in openwisp-ipam, I thought about focusing only on one model instead of showing all of them for brevity.

Added the line `@admin.register(IpAddress)`; What do you think about that?

Final result:
```
If you need to introduce significant changes and/or you don't want to resort to
monkey patching, you can proceed as follows, here we make changes in the IpAddressAdmin:

.. code-block:: python
    from django.contrib import admin
    from openwisp_ipam.admin import IpAddressAdmin as BaseIpAddressAdmin
    from swapper import load_model

    IpAddress = load_model('openwisp_ipam', 'IpAddress')

    admin.site.unregister(IpAddress)

    @admin.register(IpAddress)
    class IpAddressAdmin(BaseIpAddressAdmin):
        # add your changes here
```

Reply:
Have you ever been a coach for events like Djangogirls?
I've been a few times.
People read the docs and while they do they copy the examples verbatim and paste them into their editors.
We do it as well right? I think we should do a bit of effort to make it easier for them.
We can sacrifice brevity for the sake of completeness.
For you it's obvious that the same pattern can be applied to other models, but to someone which knows nothing about django and about the module is approaching, may be totally obscure.

Reply:
Okay, in thatcase, the only change is adding information about registration `@admin.register(IpAddress)`